public_html/wp-content/mu-plugins/0-error-handling.php

```diff
@@ -0,0 +1,169 @@
+		return false;
+	}
+
+	// Max file length for ubuntu system is 255.
+	$err_key = substr( base64_encode("$file-$line-$err_no" ), -254 );
+
+	$error_file = ERROR_RATE_LIMITING_DIR . "/$err_key";
+
+	$text = '';
+
+	$data = array(
+		'last_reported_at' => time(),
+		'error_count'      => 0, // since last reported.
+	);
+
+	if ( ! file_exists( $error_file ) ) {
+		$text = 'Error occured. ';
+		file_put_contents( $error_file, wp_json_encode( $data ) );
+	} else {
+		$data                 = json_decode( file_get_contents( $error_file ), true );
+		$data['error_count'] += 1;
+		$time_elasped         = time() - $data['last_reported_at'];
+
+		if ( $time_elasped > 600 ) {
+			$text                     = "Still happening. Happened ${data['error_count']} time(s) since last reported. ";
+			$data['last_reported_at'] = time();
+			$data['error_count']      = 0;
+
+			file_put_contents( $error_file, wp_json_encode( $data ) );
+		} else {
+			file_put_contents( $error_file, wp_json_encode( $data ) );
+			return false;
+		}
+	}
+
+	$domain    = get_site_url();
+	$page_slug = esc_html( trim( $_SERVER['REQUEST_URI'], '/' ) );
+	$text      = $text . "Message : \"$err_msg\" occurred on \"$file:$line\" \n Domain: $domain \n Page: $page_slug \n Error type: $err_no";
+
+	$message = array(
+		'fallback'    => $text,
+		'color'       => '#ff0000',
+		'pretext'     => "Error on \"$file:$line\" ",
+		'author_name' => $domain,
+		'text'        => $text,
+	);
+
+	$send = new \Dotorg\Slack\Send( SLACK_ERROR_REPORT_URL );
+	$send->add_attachment( $message );
+
+	if ( 'production' === WORDCAMP_ENVIRONMENT ) {
+		$send->send( WORDCAMP_LOGS_SLACK_CHANNEL );
+	} else {
+		$send->send( SANDBOX_SLACK_USERNAME );
+	}
+
+	return false;
+}
+
+/**
+ * Shutdown handler for catching fatal errors and sending them to Slack.
+ *
+ * Some error types cannot be handled directly by a custom error handler. However, we can catch them during shutdown
+ * and redirect them to the custom handler callback.
+ */
+function send_fatal_to_slack() {
+	$error = error_get_last();
+
+	// See https://secure.php.net/manual/en/function.set-error-handler.php
+	$unhandled_error_types = [ E_ERROR, E_PARSE, E_CORE_ERROR, E_CORE_WARNING, E_COMPILE_ERROR, E_COMPILE_WARNING ];
+
+	if ( ! empty( $error ) && in_array( $error['type'], $unhandled_error_types, true ) ) {
+		send_error_to_slack( $error['type'], $error['message'], $error['file'], $error['line'] );
+	}
+}
+
+/**
+ * Check and create filesystem dirs to manage rate limiting in error handling.
+ * For legacy bugs we are doing rate limiting via filesystem. We would be investigating to see if we can instead use memcache to rate limit sometime in the future.
+ *
+ * @return bool Return true if file permissions etc are present
+ */
+function init_error_handling() {
+	if ( ! file_exists( ERROR_RATE_LIMITING_DIR ) ) {
+		mkdir( ERROR_RATE_LIMITING_DIR );
```

Comment:
`wp_mkdir_p()` would be a best practice, but probably not helpful in this case, since `/tmp` should always exist and we wouldn't have permission to create it if it didn't.